src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs

```diff
@@ -1294,7 +1294,8 @@ private static T SerializeAndDeserialize<T>(T value, string baseline, DataContra
             if (!result.Equal && !skipStringCompare)
             {
                 Console.WriteLine(result.ErrorMessage);
-                throw new Exception(string.Format("Test failed for input : {0}", value));
+                Console.WriteLine(string.Format("Test failed for input : {0}", value));
+                Assert.StrictEqual(baseline, actualOutput);
```

Comment:
I'm unclear as to why Assert.StrictEqual is being used here.  Is it possible baseline will equal actualOutput?  If yes, then the test may actually end up passing even though we just wrote out that it failed.  And if no, wouldn't this whole if block just be an Assert.True(...) with the if condition as the condition and then all of this text as the assertion message?  Same question for the other two cases of it in this PR.